File: frontend/types/screening.ts
export interface QualityScore {
  overall_quality_score: number;
  sharpness_score: number;
  illumination_score: number;
  contrast_score: number;
  status: string;
  is_gradable: boolean;
}

export interface PredictionDetail {
  grade: number;
  stage: string;
  confidence: number;
  is_referable: boolean;
  clinical_urgency: string;
  probabilities: Record<string, number>;
}

export interface ExplainabilityDetail {
  dominant_quadrant: string;
  quadrants: {
    macula_fovea: number;
    superior_temporal: number;
    inferior_temporal: number;
    superior_nasal: number;
    inferior_nasal: number;
  };
}

export interface MultiModalImages {
  original: string;
  preprocessed: string;
  gradcam: string;
  structures: string;
}

export interface ScreeningResult {
  id: string;
  status: string;
  patient_id: string;
  eye: string;
  age?: number;
  created_at: string;
  quality: QualityScore;
  prediction: PredictionDetail;
  explainability: ExplainabilityDetail;
  images: MultiModalImages;
  report_download_url?: string;
  disclaimer: string;
}

export interface HistoryItem {
  id: string;
  patient_id: string;
  eye: string;
  age?: number;
  predicted_stage: string;
  predicted_grade: number;
  confidence: number;
  is_referable: boolean;
  quality_score: number;
  report_url?: string;
  report_download_url?: string;
  created_at: string;
}

// Aliases for compatibility
export type QualityAssessment = QualityScore;
export type PredictionResult = PredictionDetail;
export type ExplainabilityData = ExplainabilityDetail;

File: backend/app/schemas/screening.py
from typing import Dict, Any, Optional, List
from datetime import datetime
from pydantic import BaseModel, Field

class PatientInfo(BaseModel):
    patient_id: str = Field(default="ANON_PATIENT", description="Clinical patient identifier")
    eye: str = Field(default="OD", description="Eye side: OD (Right) or OS (Left)")
    age: Optional[int] = Field(default=None, description="Patient age in years")

class QualityScoreResponse(BaseModel):
    overall_quality_score: float
    sharpness_score: float
    illumination_score: float
    contrast_score: float
    status: str
    is_gradable: bool

class PredictionDetail(BaseModel):
    grade: int
    stage: str
    confidence: float
    is_referable: bool
    clinical_urgency: str
    probabilities: Dict[str, float]

class ExplainabilityDetail(BaseModel):
    dominant_quadrant: str
    quadrants: Dict[str, float]

class MultiModalImages(BaseModel):
    original: str
    preprocessed: str
    gradcam: str
    structures: str

class ScreeningResponse(BaseModel):
    id: str
    status: str
    patient_id: str
    eye: str
    age: Optional[int]
    created_at: str
    quality: QualityScoreResponse
    prediction: PredictionDetail
    explainability: ExplainabilityDetail
    images: MultiModalImages
    report_download_url: Optional[str] = None
    disclaimer: str

class ScreeningHistoryItem(BaseModel):
    id: str
    patient_id: str
    eye: str
    age: Optional[int]
    predicted_stage: str
    predicted_grade: int
    confidence: float
    is_referable: bool
    quality_score: float
    created_at: datetime

    class Config:
        from_attributes = True
